# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/ipc/RemoteAccessibleBase.h
# Commit: 3cfe5bd5d084
# Full Hash: 3cfe5bd5d084baf18f5d95b0e879f6fc179fac34
# Author: Morgan Reschenberg <mreschenberg@mozilla.com>
# Date: 2021-10-06 04:09:04
# Regressor Bug: 1726227
# File Overlap Count: 1
# Description:
#   Bug 1726227: Cache parent-relative accessible bounds, resolution in parent process r=Jamie,eeejay
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D123399
# ==============================================================================

diff -r 76a965804210 -r 3cfe5bd5d084 accessible/ipc/RemoteAccessibleBase.h
--- a/accessible/ipc/RemoteAccessibleBase.h	Tue Oct 05 20:24:03 2021 +0000
+++ b/accessible/ipc/RemoteAccessibleBase.h	Tue Oct 05 20:44:46 2021 +0000
@@ -175,6 +175,8 @@
   virtual double MaxValue() const override;
   virtual double Step() const override;
 
+  virtual nsIntRect Bounds() const override;
+
   /**
    * Allow the platform to store a pointers worth of data on us.
    */
@@ -234,6 +236,7 @@
 
  protected:
   void SetParent(Derived* aParent);
+  Maybe<nsRect> RetrieveCachedBounds() const;
 
  private:
   uintptr_t mParent;